### ISSUE TITLE
[feat] evaluate wagmi/useDapp/custom solution for handling contract transactions

### ISSUE THREAD

**Issue by anthonymartin** [popcorndao/sweet-caramel#1110]
**Description**

We would like to simplify contract interaction and centralize certain logic to handle pre & post transaction submission. Currently if we wanted to handle checks before a transaction is made, we would need to update every single file where a transaction is submitted. 

We need to evaluate the best option among these: wagmi, useDapp, custom. The preference would be to use wagmi, considering it's adoption is skyrocketing.

**Acceptance Criteria**

- [x] identify if it is possible to simply use wagmi/useDapp for transactions without having to change the web3-onboard library.
- [x] create proof of concept with choosen solution

**Comment by anthonymartin** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Default-603eb49d1a9a0b0011b49b36/issues/popcorndao/sweet-caramel/1110?planning-poker) with ZenHub @Ajiva-D @Amirjab21 @arne-z @ecmendenhall @K-G-J

**Comment by arne-z** [comment]
https://docs.google.com/document/d/1LRMYGt58lsVrD9Zx_7YGiU3vhPxYuh--R1RO07MA_VM/edit?usp=sharing

**Comment by RedVeil** [comment]
@anthonymartin @arne-z 
WAGMI sounds like a great solution in theory but would require quite the refactor. It might make sense to think about it when we start with our refactoring but only if it would save a sizable amount of time in the future. For now this change seems way to invasive. 
Another issue i have with WAGMI is looking longer term the support for multiple networks at once. We would like to show multiple contracts from multiple chains in one unified page. This is only possible if we can use multiple RPC-Provider in parallel. It sounds like this is not supported by default and might need some forking or conversations with the wagmi-devs to find out how to make this work.
Regardless, WAGMI sounds like way to much work for our current issue.
 
The integration of useDapp seems manageable. Sure we have to find out how to await transactions or if thats already handled internally by useDapp but that seems feasible. I would also ask here if we can run multiple hooks with different RPC_PROVIDER in parallel. @arne-z Do you know that?
